design/PRODUCER-MIDDLEWARE.md

```diff
@@ -3,6 +3,40 @@
 The producer is responsible for:
 
 - serving the HTTP requests for pages of events
-- output caching of completed pages
+- output/response caching of completed pages
 - collecting of meta data about usage (see persistence design for details)
-- only serve the event feed to authorized consumers
\ No newline at end of file
+- only serve the event feed to authorized consumers
+
+## How
+
+[See caching discussion here](https://github.com/dburriss/event-feed/issues/2).
+ 
+ ```mermaid
+sequenceDiagram
+
+    participant DB as Event Data Store
+    participant S as Consumer Cache
+    participant P as Producer
+    participant C as Consumer
+
+    C ->>+ P: Fetch meta data
+    P -> DB: Retrieve meta data from DB
+    P -->>- C: Return meta data
+    
+    C ->>+ P: Fetch a page
+    P ->> S: Try fetch page
+    alt Page is in cache
```

Comment:
The cache is responsible for returning from cache or fetching the page from storage. I think this is subtly different from the diagram which shows the Producer querying the cache and on miss then querying the Event Data Store.

Reply:
In the sequence Cache is just Memory or Redis or MS-SQL DB. IT cannot do anything but respond with what asked for. Was not thinking of it as a layer of code in the lib. I tried making the actor glyph a DB shape but mermaid does not support that yet like platuml.

Reply:
Okay got it, I was mistakenly thinking of it in terms of components in the service, not the overall system. But that makes sense indeed.